### PR TITLE
Experiments optimisations

### PR DESCRIPTION
**Pull request by RealOrangeOne**
#### When applied this MR will...

Improve the performance of 3 key sections:
- Participant list
- All "Data" list screens
- Data export

#### Please provide detail on the technical changes, if relevant

The participant list hasn't been improved much, only tiny bits. This is because most of the issues are coming from the form fields, which are obscured behind the factory, and would take longer than the time allocated. The main speed change was f5b9a33, which removed the unnecessary fields. For reasons I don't understand, removing these lines makes it much faster, even though the resulting HTML is trivial (and results in no extra DB hits).

The optimisation for the data screens was a simple `select_related` e73b1b2

The data export optimisation was in 2 parts. The first was adding some `select_related` to cover the fields missed. The second was a small optimisation to how the CSVs were generated, resulting in less stored in memory and hopefully reduce memory usage. It is far from perfect, but hopefully improves things slightly. There are avenues for future improvement, though.

### CODE REVIEW

File: flare_portal/experiments/templates/experiments/participant_list.html [hunk 1]
```diff
@@ -113,11 +113,9 @@ <h3 class="card-title">Participants</h3>
                                                 <span class="status-icon {% if form.instance.has_been_rejected %}bg-success{% else %}bg-danger{% endif %}"></span> {{ form.instance.has_been_rejected|yesno|title }}
                                             </td>
 
-                                            {% if experiment.voucher_pool %} <td>{{ form.instance.get_voucher_status }}</td> {% endif %}
+                                            {% if experiment.voucher_pool_id %} <td>{{ form.instance.get_voucher_status }}</td> {% endif %}
 
                                             <td>
-                                                {{ form.id }}
```
Comment (Morsey187) [inline flare_portal/experiments/templates/experiments/participant_list.html:119]: **Issue** Sadly these two fields are required for the "add participants" -> "save changes" form to to work

**Review by Morsey187 — COMMENTED** [review]
This is great work, the changes to the file export classes has more than halfed the download time for me locally. 

This commit will have to be reverted [f5b9a33](https://github.com/flare-kcl/flare-portal/commit/f5b9a335306891d849a19e724c90cf0a35e5e7b9) sadly, as mentioned in the review, and yes theres definitely avenues for future improvement in regards to the particpant list, which will have to be done as a seperate piece of work.

File: flare_portal/experiments/templates/experiments/participant_list.html (hunk 1 above)
Comment (RealOrangeOne) [reply flare_portal/experiments/templates/experiments/participant_list.html:119]: Really? From my (albeit brief) testing, saving appeared to work fine. It doesn't look like the forms use those fields anyway, and the frontend JS only applies to the newly created rows, not existing ones.

**Review by mixxorz — APPROVED** [review]
This looks good to me. Thanks Jake!

File: flare_portal/experiments/templates/experiments/participant_list.html (hunk 1 above)
Comment (Morsey187) [reply flare_portal/experiments/templates/experiments/participant_list.html:119]: Just to highlight, heres an example of this issue triggering an error on staging:
 
 
Error on sentry: https://sentry.io/organizations/torchbox/issues/2387572326/?project=5479528&query=is%3Aunresolved

`inlineformset_factory` Will still expect existing forms to submit their fields so {{ form.id }} is still required as this is the paricipant id in this case.
You may have avoided this error by creating a new participant list from scratch and not trying to save the form after a page refresh with new fields.

Heres the formset:
```
ParticipantFormSet = inlineformset_factory(
    Experiment, Participant, fields=["participant_id"], extra=0
)
```

**Pull request merged**